game-core/src/main/java/games/strategy/engine/data/gameparser/GameParser.java

```diff
@@ -395,149 +371,118 @@ private void parseResources(final Element root) throws GameParseException {
-    if (getSingleChildOptional("relationshipInitialize", root).orElse(null) == null) {
-      final RelationshipTracker relationshipTracker = data.getRelationshipTracker();
-      final RelationshipTypeList relationshipTypeList = data.getRelationshipTypeList();
-      // iterate through all players to get known allies and enemies
-      for (final GamePlayer currentPlayer : players) {
-        // start with all players as enemies
-        // start with no players as allies
-        final Set<GamePlayer> allies = allianceTracker.getAllies(currentPlayer);
-        final Set<GamePlayer> enemies = new HashSet<>(players);
-        enemies.removeAll(allies);
-
-        // remove self from enemies list (in case of free-for-all)
-        enemies.remove(currentPlayer);
-        // remove self from allies list (in case you are a member of an alliance)
-        allies.remove(currentPlayer);
-        // At this point enemies and allies should be set for this player.
-        for (final GamePlayer alliedPLayer : allies) {
-          relationshipTracker.setRelationship(
-              currentPlayer, alliedPLayer, relationshipTypeList.getDefaultAlliedRelationship());
-        }
-        for (final GamePlayer enemyPlayer : enemies) {
-          relationshipTracker.setRelationship(
-              currentPlayer, enemyPlayer, relationshipTypeList.getDefaultWarRelationship());
-        }
+    final RelationshipTracker relationshipTracker = data.getRelationshipTracker();
+    final RelationshipTypeList relationshipTypeList = data.getRelationshipTypeList();
+    // iterate through all players to get known allies and enemies
+    for (final GamePlayer currentPlayer : players) {
+      // start with all players as enemies
+      // start with no players as allies
+      final Set<GamePlayer> allies = allianceTracker.getAllies(currentPlayer);
+      final Set<GamePlayer> enemies = new HashSet<>(players);
+      enemies.removeAll(allies);
+
+      // remove self from enemies list (in case of free-for-all)
+      enemies.remove(currentPlayer);
+      // remove self from allies list (in case you are a member of an alliance)
+      allies.remove(currentPlayer);
+      // At this point enemies and allies should be set for this player.
+      for (final GamePlayer alliedPLayer : allies) {
+        relationshipTracker.setRelationship(
+            currentPlayer, alliedPLayer, relationshipTypeList.getDefaultAlliedRelationship());
+      }
+      for (final GamePlayer enemyPlayer : enemies) {
+        relationshipTracker.setRelationship(
+            currentPlayer, enemyPlayer, relationshipTypeList.getDefaultWarRelationship());
       }
     }
   }
 
-  private void parseRelationInitialize(final List<Element> relations) throws GameParseException {
-    if (!relations.isEmpty()) {
+  private void parseRelationInitialize(final Initialize.RelationshipInitialize relations)
+      throws GameParseException {
+    if (!relations.getRelationships().isEmpty()) {
```

Comment:
I wonder where this check comes from. If the list is empty the for loop shouldn't iterate a single time and getRelationShipTracker doesn't look like it would throw an exception